src/httpd_jsonapi.c

```diff
@@ -2063,7 +2064,9 @@ queue_item_to_json(struct db_queue_item *queue_item, char shuffle)
 
   if (queue_item->artwork_url
       && (strncmp(queue_item->artwork_url, "http://", strlen("http://")) == 0
-	  || strncmp(queue_item->artwork_url, "https://", strlen("https://")) == 0))
+	  || strncmp(queue_item->artwork_url, "https://", strlen("https://")) == 0) && (
+      (ext = strrchr(queue_item->artwork_url, '.')) && (strcmp(ext, ".jpg") == 0 || strcmp(ext, ".png") == 0 )
```

Comment:
Ooof, we really shouldn't have url's in artwork_url that aren't artwork_url's... I guess we are reading streamurl straight into artwork_url now, so I'm thinking the extension check should be there instead.

Reply:
Are you suggesting that somewhere in artwork.c it updates the artwork_url value in the database once it finds the correct image URL?